Добрый день. Проект по аналитике глобальных данных продаж. Из данного проекта собираюсь....

In [67]:
import pandas as pd

In [13]:
pd.set_option('display.max_columns', None)

In [26]:
# Первым делом открываю свой файл и смотрю первые строки, изучаю о чем столбцы,
# какие значения в строках

df = pd.read_csv(r"C:\Users\nurik\Desktop\курс Python\Global Superstore Data\Data_Analysis\superstore.csv")
df.head(15)

Category         City        Country Customer.ID  \
0   Office Supplies  Los Angeles  United States   LS-172304   
1   Office Supplies  Los Angeles  United States   MV-174854   
2   Office Supplies  Los Angeles  United States   CS-121304   
3   Office Supplies  Los Angeles  United States   CS-121304   
4   Office Supplies  Los Angeles  United States   AP-109154   
5   Office Supplies  Los Angeles  United States   JF-154904   
6   Office Supplies  Los Angeles  United States   WB-218504   
7   Office Supplies  Los Angeles  United States   JA-159704   
8   Office Supplies  Los Angeles  United States   SP-209204   
9   Office Supplies  Los Angeles  United States   RL-196154   
10  Office Supplies  Los Angeles  United States   KD-163454   
11  Office Supplies  Los Angeles  United States   JM-161954   
12  Office Supplies  Los Angeles  United States   JM-161954   
13  Office Supplies  Los Angeles  United States   RD-195854   
14  Office Supplies  Los Angeles  United States   AB-101654   

         Customer.Name  Discount Market  记录数               Order.Date  \
0     Lycoris Saunders       0.0     US    1  2011-01-07 00:00:00.000   
1        Mark Van Huff       0.0     US    1  2011-01-21 00:00:00.000   
2         Chad Sievert       0.0     US    1  2011-08-05 00:00:00.000   
3         Chad Sievert       0.0     US    1  2011-08-05 00:00:00.000   
4       Arthur Prichep       0.0     US    1  2011-09-29 00:00:00.000   
5         Jeremy Farry       0.0     US    1  2011-10-19 00:00:00.000   
6        William Brown       0.0     US    1  2011-11-04 00:00:00.000   
7         Joseph Airdo       0.0     US    1  2011-11-12 00:00:00.000   
8         Susan Pistek       0.0     US    1  2011-11-22 00:00:00.000   
9            Rob Lucas       0.0     US    1  2011-12-05 00:00:00.000   
10    Katherine Ducich       0.0     US    1  2011-12-13 00:00:00.000   
11  Justin MacKendrick       0.0     US    1  2012-02-15 00:00:00.000   
12  Justin MacKendrick       0.0     US    1  2012-02-15 00:00:00.000   
13            Rob Dowd       0.0     US    1  2012-02-16 00:00:00.000   
14         Alan Barnes       0.0     US    1  2012-03-28 00:00:00.000   

          Order.ID Order.Priority       Product.ID  \
0   CA-2011-130813           High  OFF-PA-10002005   
1   CA-2011-148614         Medium  OFF-PA-10002893   
2   CA-2011-118962         Medium  OFF-PA-10000659   
3   CA-2011-118962         Medium  OFF-PA-10001144   
4   CA-2011-146969           High  OFF-PA-10002105   
5   CA-2011-117317       Critical  OFF-PA-10004519   
6   CA-2011-125829         Medium  OFF-PA-10000223   
7   CA-2011-151295           High  OFF-PA-10001947   
8   CA-2011-135090           High  OFF-PA-10002245   
9   CA-2011-133830         Medium  OFF-PA-10001363   
10  CA-2011-113579           High  OFF-PA-10001457   
11  CA-2012-116092         Medium  OFF-PA-10004285   
12  CA-2012-116092         Medium  OFF-PA-10000477   
13  CA-2012-121776           High  OFF-PA-10001970   
14  CA-2012-131534         Medium  OFF-PA-10000743   

                                         Product.Name   Profit  Quantity  \
0                                           Xerox 225   9.3312         3   
1           Wirebound Service Call Books, 5 1/2" x 4"   9.2928         2   
2   Adams Phone Message Book, Professional, 400 Me...   9.8418         3   
3                                          Xerox 1913  53.2608         2   
4                                           Xerox 223   3.1104         1   
5     Spiral Phone Message Books with Labels by Adams   6.5856         3   
6                                          Xerox 2000   9.3312         3   
7                                          Xerox 1974   5.8604         2   
8                                          Xerox 1895  24.2190         9   
9                                          Xerox 1933  23.0864         4   
10                         White GlueTop Scratch Pads  41.5104         6   
11                                         Xerox 1959   

In [25]:
# Проверка информации о файле показывает мне, какие есть основные столбцы, 
# просмотрев которые, я могу подметить, что на основе этих данных можно сделать 
# действительно большой анализ. Информация показывает мне, что основные данные
# это 'object', а памяти файл занимает 10+ MB
# всего 51290 строк и 27 столбцов

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 27 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Category        51290 non-null  object 
 1   City            51290 non-null  object 
 2   Country         51290 non-null  object 
 3   Customer.ID     51290 non-null  object 
 4   Customer.Name   51290 non-null  object 
 5   Discount        51290 non-null  float64
 6   Market          51290 non-null  object 
 7   记录数             51290 non-null  int64  
 8   Order.Date      51290 non-null  object 
 9   Order.ID        51290 non-null  object 
 10  Order.Priority  51290 non-null  object 
 11  Product.ID      51290 non-null  object 
 12  Product.Name    51290 non-null  object 
 13  Profit          51290 non-null  float64
 14  Quantity        51290 non-null  int64  
 15  Region          51290 non-null  object 
 16  Row.ID          51290 non-null  int64  
 17  Sales           51290 non-null 

In [28]:
# Данный код позволил мне обнаружить, что в этом датасете нет ни единого
# пропуска данных, что несомненно упростит работу и анализ информации

df_null = df.isnull().sum()
df_null

Category          0
City              0
Country           0
Customer.ID       0
Customer.Name     0
Discount          0
Market            0
记录数               0
Order.Date        0
Order.ID          0
Order.Priority    0
Product.ID        0
Product.Name      0
Profit            0
Quantity          0
Region            0
Row.ID            0
Sales             0
Segment           0
Ship.Date         0
Ship.Mode         0
Shipping.Cost     0
State             0
Sub.Category      0
Year              0
Market2           0
weeknum           0
dtype: int64

In [31]:
# На данном этапе сделал интересное замечание, что в графе
# "Sales" есть минимиальное значение 0, что хотелось бы посмотреть
# Также, при изучении аутпута вижу, что максимальным значением "Quantity"
# является 14, что может дать информацию о востребованном товаре или
# качественном потребителе

df.describe()

Discount      记录数        Profit      Quantity       Row.ID  \
count  51290.000000  51290.0  51290.000000  51290.000000  51290.00000   
mean       0.142908      1.0     28.610982      3.476545  25645.50000   
std        0.212280      0.0    174.340972      2.278766  14806.29199   
min        0.000000      1.0  -6599.978000      1.000000      1.00000   
25%        0.000000      1.0      0.000000      2.000000  12823.25000   
50%        0.000000      1.0      9.240000      3.000000  25645.50000   
75%        0.200000      1.0     36.810000      5.000000  38467.75000   
max        0.850000      1.0   8399.976000     14.000000  51290.00000   

              Sales  Shipping.Cost          Year       weeknum  
count  51290.000000   51290.000000  51290.000000  51290.000000  
mean     246.498440      26.375818   2012.777208     31.287112  
std      487.567175      57.296810      1.098931     14.429795  
min        0.000000       0.002000   2011.000000      1.000000  
25%       31.000000       2.610000   2012.000000     20.000000  
50%       85.000000       7.790000   2013.000000     33.000000  
75%      251.000000      24.450000   2014.000000     44.000000  
max    22638.000000     933.570000   2014.000000     53.000000

In [72]:
# Проверка уникальных значений в каждом столбце.
# Просматривает информация, например, что информация собрана за 4 года
# Также, видно, что продажи проводились на 7 маркетах в 13 регионах мира
# Всего имеется 4 класса доставки и 3 сегмента покупателя с 4 уровнями
# приоритетности. Нашел странную 53 неделю в году, проверяю

df.nunique()

Category              3
City               3636
Country             147
Customer.ID        4873
Customer.Name       795
Discount             27
Market                7
Order.Date         1430
Order.ID          25035
Order.Priority        4
Product.ID        10292
Product.Name       3788
Profit            24575
Quantity             14
Region               13
Sales              2246
Segment               3
Ship.Date          1464
Ship.Mode             4
Shipping.Cost     16877
State              1094
Sub.Category         17
Year                  4
Market2               6
weeknum              53
dtype: int64

In [79]:
# Проверяю количество уникальных значений в столбце "weeknum"
# В году их должно быть 52
df['weeknum'].unique()

array([40, 42, 39, 10, 26, 51, 44, 25, 21, 20, 27, 50,  3, 17, 19, 14, 22,
       35, 41, 43, 11, 45, 23, 30, 37,  9,  7,  4, 52, 32, 49, 48, 24, 36,
       33, 38, 12, 53,  2, 46, 29, 28, 47, 31, 34,  5, 16, 15,  8,  6, 18,
       13,  1])

In [80]:
# После вывода всех 53 недель заказов, обнаружил, что на самом деле
# все корректно, в 2012 году 31 декабря стало еще одной неделей декабря (ПН)
# что перенесло даты до 6 января в прошлый год пятую неделю

df.loc[df['weeknum'] == 53]

Category           City             Country Customer.ID  \
Row.ID                                                                   
123           Furniture      Carrefour               Haiti   PB-191503   
124          Technology      Carrefour               Haiti   PB-191503   
216          Technology  Santo Domingo  Dominican Republic   RR-195253   
217           Furniture  Santo Domingo  Dominican Republic   RR-195253   
218          Technology  Santo Domingo  Dominican Republic   RR-195253   
...                 ...            ...                 ...         ...   
51138   Office Supplies          Adana              Turkey    LO-71702   
51139   Office Supplies          Adana              Turkey    LO-71702   
51241   Office Supplies         Ibadan             Nigeria    NF-83851   
51242   Office Supplies         Ibadan             Nigeria    NF-83851   
51243   Office Supplies         Ibadan             Nigeria    NF-83851   

           Customer.Name  Discount  Market               Order.Date  \
Row.ID                                                                
123         Philip Brown       0.4   LATAM  2011-12-29 00:00:00.000   
124         Philip Brown       0.7   LATAM  2011-12-29 00:00:00.000   
216            Rick Reed       0.2   LATAM  2011-12-28 00:00:00.000   
217            Rick Reed       0.2   LATAM  2011-12-28 00:00:00.000   
218            Rick Reed       0.2   LATAM  2011-12-28 00:00:00.000   
...                  ...       ...     ...                      ...   
51138         Lori Olson       0.6    EMEA  2011-12-26 00:00:00.000   
51139         Lori Olson       0.6    EMEA  2011-12-26 00:00:00.000   
51241   Natalie Fritzler       0.7  Africa  2014-12-30 00:00:00.000   
51242   Natalie Fritzler       0.7  Africa  2014-12-30 00:00:00.000   
51243   Natalie Fritzler       0.7  Africa  2014-12-30 00:00:00.000   

              Order.ID Order.Priority        Product.ID  \
Row.ID                                                    
123     US-2011-138842           High   FUR-BO-10003103   
124     US-2011-138842           High   TEC-MA-10001290   
216     US-2011-128776       Critical   TEC-PH-10004871   
217     US-2011-128776       Critical   FUR-CH-10004725   
218     US-2011-128776       Critical   TEC-AC-10001491   
...                ...            ...               ...   
51138     TU-2011-8780         Medium  OFF-SME-10004702   
51139     TU-2011-8780         Medium  OFF-AVE-10004828   
51241     NI-2014-3370         Medium  OFF-IBI-10004916   
51242     NI-2014-3370         Medium  OFF-SME-10001853   
51243     NI-2014-3370         Medium  OFF-BIN-10004512   

                                            Product.Name   Profit  Quantity  \
Row.ID                                                                        
123                       Dania Floating Shelf Set, Pine  -95.600         4   
124                             Panasonic Printer, White -127.160         2   
216                           Motorola Headset, Cordless  -14.344         2   
217                   Office Star Bag Chairs, Set of Two  -10.548         9   
218                       Enermax Flash Drive, Bluetooth  -23.580         5   
...                                                  ...      ...       ...   
51138                        Smead File Cart, Industrial  -41.100         1   
51139   Avery Removable Labels, Laser Printer Compatible   -8.628         2   
51241                      Ibico Binder Covers, Recycled   -4.647         1   
51242                                Smead Folders, Blue  -10.173         1   
51243                    Binney & Smith Pens, Easy-Erase  -51.282         6   

           Region  Sales    Segment                Ship.Date       Ship.Mode  \
Row.ID                                                                         
123     Caribbean    273   Consumer  2012-01-02 00:00:00.000  Standard Class   
124     Caribbean    106   Consumer  2012-01-02 00:00:00.000  Standard Class   
216     Cari

In [40]:
# В данном блоке кода установил столбец "Row.ID" как индекс
# а также сделал сортировку от первого до последнего
# Это сделает таблицу более приятной и воспринимаемой для глаз

df = df.set_index('Row.ID')
df = df.sort_index()
df

Category           City   Country Customer.ID  \
Row.ID                                                         
1       Office Supplies    Mexico City    Mexico   SC-205753   
2             Furniture  Dos Quebradas  Colombia   KW-165703   
3             Furniture  Dos Quebradas  Colombia   KW-165703   
4       Office Supplies  Dos Quebradas  Colombia   KW-165703   
5       Office Supplies  Dos Quebradas  Colombia   KW-165703   
...                 ...            ...       ...         ...   
51286   Office Supplies       Budapest   Hungary    NF-83853   
51287        Technology       Budapest   Hungary    NF-83853   
51288   Office Supplies       Budapest   Hungary    NF-83853   
51289   Office Supplies       Budapest   Hungary    NF-83853   
51290   Office Supplies        Baghdad      Iraq   SZ-100352   

           Customer.Name  Discount Market  记录数               Order.Date  \
Row.ID                                                                    
1           Sonia Cooley       0.0  LATAM    1  2014-10-02 00:00:00.000   
2         Kelly Williams       0.0  LATAM    1  2012-10-15 00:00:00.000   
3         Kelly Williams       0.0  LATAM    1  2012-10-15 00:00:00.000   
4         Kelly Williams       0.0  LATAM    1  2012-10-15 00:00:00.000   
5         Kelly Williams       0.0  LATAM    1  2012-10-15 00:00:00.000   
...                  ...       ...    ...  ...                      ...   
51286   Natalie Fritzler       0.0   EMEA    1  2012-09-05 00:00:00.000   
51287   Natalie Fritzler       0.0   EMEA    1  2012-09-05 00:00:00.000   
51288   Natalie Fritzler       0.0   EMEA    1  2012-09-05 00:00:00.000   
51289   Natalie Fritzler       0.0   EMEA    1  2012-09-05 00:00:00.000   
51290         Sam Zeldin       0.0   EMEA    1  2011-08-25 00:00:00.000   

              Order.ID Order.Priority        Product.ID  \
Row.ID                                                    
1       MX-2014-143658         Medium   OFF-LA-10002782   
2       MX-2012-155047         Medium   FUR-FU-10004015   
3       MX-2012-155047         Medium   FUR-BO-10002352   
4       MX-2012-155047         Medium   OFF-BI-10004428   
5       MX-2012-155047         Medium   OFF-AR-10004594   
...                ...            ...               ...   
51286     HU-2012-7730           High  OFF-AVE-10004570   
51287     HU-2012-7730           High  TEC-LOG-10004419   
51288     HU-2012-7730           High  OFF-BOS-10002705   
51289     HU-2012-7730           High  OFF-ENE-10004132   
51290     IZ-2011-5750         Medium  OFF-ACC-10002343   

                                     Product.Name  Profit  Quantity Region  \
Row.ID                                                                       
1              Hon File Folder Labels, Adjustable    4.56         3  North   
2                            Tenex Clock, Durable   90.72         8  South   
3                    Ikea 3-Shelf Cabinet, Mobile   54.08         2  South   
4                          Cardinal Binder, Clear    4.96         4  South   
5                     Sanford Canvas, Water Color   11.44         2  South   
...                                           ...     ...       ...    ...   
51286                Avery Binder Covers, Economy    3.42         1   EMEA   
51287                   Logitech Flash Drive, USB   18.42         2   EMEA   
51288            Boston Highlighters, Fluorescent   20.88         4   EMEA   
51289   Enermax Computer Printout Paper, 8.5 x 11   33.84         4   EMEA   
51290                  Acco 3-Hole Punch, Durable   12.96         1   EMEA   

        Sales      Segment                Ship.Date       Ship.Mode  \
Row.ID                                                                
1          13     Consumer  2014-10-06 00:00:00.000  Standard Class   
2         252     Consumer  2012-10-20 00:00:00.000  Standard Class   
3         193     Consumer  2012-10-20 00:00:00.000  Standard Class   
4          35     Consumer  2012-10-20 00:00:00.000  Standard Class 

Перехожу к более детальному изучению столбцов и строк в данном датасете, изучаю, какие строки или столбцы можно удалить за ненадобностью, чтобы разгрузить таблицу


In [63]:
# При обследовании таблицы через ".describe" было обнаружено, что столбец (记录数)
# имеет всего одно значение "1", проверяю сортировкой, подтверждается

df.sort_values(df.columns[7], ascending=False)

Category           City    Country Customer.ID   Customer.Name  \
Row.ID                                                                          
51290   Office Supplies        Baghdad       Iraq   SZ-100352      Sam Zeldin   
1       Office Supplies    Mexico City     Mexico   SC-205753    Sonia Cooley   
2             Furniture  Dos Quebradas   Colombia   KW-165703  Kelly Williams   
3             Furniture  Dos Quebradas   Colombia   KW-165703  Kelly Williams   
4       Office Supplies  Dos Quebradas   Colombia   KW-165703  Kelly Williams   
...                 ...            ...        ...         ...             ...   
15      Office Supplies       Brasília     Brazil   JK-156253    Jim Karlsson   
14      Office Supplies        Managua  Nicaragua   PK-189103    Paul Knutson   
13            Furniture        Managua  Nicaragua   TB-212503    Tim Brockman   
12      Office Supplies        Managua  Nicaragua   TB-212503    Tim Brockman   
11      Office Supplies      São Paulo     Brazil   DP-130003   Darren Powers   

        Discount Market  记录数               Order.Date        Order.ID  \
Row.ID                                                                  
51290        0.0   EMEA    1  2011-08-25 00:00:00.000    IZ-2011-5750   
1            0.0  LATAM    1  2014-10-02 00:00:00.000  MX-2014-143658   
2            0.0  LATAM    1  2012-10-15 00:00:00.000  MX-2012-155047   
3            0.0  LATAM    1  2012-10-15 00:00:00.000  MX-2012-155047   
4            0.0  LATAM    1  2012-10-15 00:00:00.000  MX-2012-155047   
...          ...    ...  ...                      ...             ...   
15           0.0  LATAM    1  2014-10-18 00:00:00.000  MX-2014-135706   
14           0.0  LATAM    1  2014-09-24 00:00:00.000  MX-2014-121923   
13           0.0  LATAM    1  2013-03-05 00:00:00.000  MX-2013-156335   
12           0.0  LATAM    1  2013-03-05 00:00:00.000  MX-2013-156335   
11           0.0  LATAM    1  2013-09-27 00:00:00.000  MX-2013-134096   

       Order.Priority        Product.ID  \
Row.ID                                    
51290          Medium  OFF-ACC-10002343   
1              Medium   OFF-LA-10002782   
2              Medium   FUR-FU-10004015   
3              Medium   FUR-BO-10002352   
4              Medium   OFF-BI-10004428   
...               ...               ...   
15             Medium   OFF-AR-10003914   
14                Low   OFF-EN-10004100   
13             Medium   FUR-CH-10002846   
12             Medium   OFF-FA-10002526   
11             Medium   OFF-BI-10002075   

                                     Product.Name  Profit  Quantity   Region  \
Row.ID                                                                         
51290                  Acco 3-Hole Punch, Durable   12.96         1     EMEA   
1              Hon File Folder Labels, Adjustable    4.56         3    North   
2                            Tenex Clock, Durable   90.72         8    South   
3                    Ikea 3-Shelf Cabinet, Mobile   54.08         2    South   
4                          Cardinal Binder, Clear    4.96         4    South   
...                                           ...     ...       ...      ...   
15                     Stanley Canvas, Easy-Erase   35.76         4    South   
14                   Kraft Mailers, Security-Tint   37.62         3  Central   
13                 Hon Steel Folding Chair, Black   50.48         4  Central   
12                         Accos Staples, 12 Pack    9.54         3  Central   
11      Wilson Jones Hole Reinforcements, Durable    3.48         3    South   

        Sales      Segment                Ship.Date       Ship.Mode  \
Row.ID                                                                
51290      32  Home Office  2011-08-29 00:00:00.000  Standard Class   
1          13     Consumer  2014-10-06 00:00:00.000  Standard Class   
2         252     Consumer  2012-10-20 00:00:00.000  Standard Class   
3         193     Consumer  2012-10-20 00:00:00.000  St

In [64]:
# Также решил проверить количество уникальных записей в данном столбце, для 
# более точного подтверждения, чтобы не осталось сомнений о бесполезности
# данного столбца

df[df.columns[7]].unique()

array([1])

In [66]:
# В данном коде избавился от лишнего столбца (记录数), который означал "Количество записей"
# Столбец бесполезный, ведь каждая строка уникальна, нет ни одной строки,
# полностью повторяющейся. 

df = df.drop(df.columns[7], axis=1)
df

Category           City   Country Customer.ID  \
Row.ID                                                         
1       Office Supplies    Mexico City    Mexico   SC-205753   
2             Furniture  Dos Quebradas  Colombia   KW-165703   
3             Furniture  Dos Quebradas  Colombia   KW-165703   
4       Office Supplies  Dos Quebradas  Colombia   KW-165703   
5       Office Supplies  Dos Quebradas  Colombia   KW-165703   
...                 ...            ...       ...         ...   
51286   Office Supplies       Budapest   Hungary    NF-83853   
51287        Technology       Budapest   Hungary    NF-83853   
51288   Office Supplies       Budapest   Hungary    NF-83853   
51289   Office Supplies       Budapest   Hungary    NF-83853   
51290   Office Supplies        Baghdad      Iraq   SZ-100352   

           Customer.Name  Discount Market               Order.Date  \
Row.ID                                                               
1           Sonia Cooley       0.0  LATAM  2014-10-02 00:00:00.000   
2         Kelly Williams       0.0  LATAM  2012-10-15 00:00:00.000   
3         Kelly Williams       0.0  LATAM  2012-10-15 00:00:00.000   
4         Kelly Williams       0.0  LATAM  2012-10-15 00:00:00.000   
5         Kelly Williams       0.0  LATAM  2012-10-15 00:00:00.000   
...                  ...       ...    ...                      ...   
51286   Natalie Fritzler       0.0   EMEA  2012-09-05 00:00:00.000   
51287   Natalie Fritzler       0.0   EMEA  2012-09-05 00:00:00.000   
51288   Natalie Fritzler       0.0   EMEA  2012-09-05 00:00:00.000   
51289   Natalie Fritzler       0.0   EMEA  2012-09-05 00:00:00.000   
51290         Sam Zeldin       0.0   EMEA  2011-08-25 00:00:00.000   

              Order.ID Order.Priority        Product.ID  \
Row.ID                                                    
1       MX-2014-143658         Medium   OFF-LA-10002782   
2       MX-2012-155047         Medium   FUR-FU-10004015   
3       MX-2012-155047         Medium   FUR-BO-10002352   
4       MX-2012-155047         Medium   OFF-BI-10004428   
5       MX-2012-155047         Medium   OFF-AR-10004594   
...                ...            ...               ...   
51286     HU-2012-7730           High  OFF-AVE-10004570   
51287     HU-2012-7730           High  TEC-LOG-10004419   
51288     HU-2012-7730           High  OFF-BOS-10002705   
51289     HU-2012-7730           High  OFF-ENE-10004132   
51290     IZ-2011-5750         Medium  OFF-ACC-10002343   

                                     Product.Name  Profit  Quantity Region  \
Row.ID                                                                       
1              Hon File Folder Labels, Adjustable    4.56         3  North   
2                            Tenex Clock, Durable   90.72         8  South   
3                    Ikea 3-Shelf Cabinet, Mobile   54.08         2  South   
4                          Cardinal Binder, Clear    4.96         4  South   
5                     Sanford Canvas, Water Color   11.44         2  South   
...                                           ...     ...       ...    ...   
51286                Avery Binder Covers, Economy    3.42         1   EMEA   
51287                   Logitech Flash Drive, USB   18.42         2   EMEA   
51288            Boston Highlighters, Fluorescent   20.88         4   EMEA   
51289   Enermax Computer Printout Paper, 8.5 x 11   33.84         4   EMEA   
51290                  Acco 3-Hole Punch, Durable   12.96         1   EMEA   

        Sales      Segment                Ship.Date       Ship.Mode  \
Row.ID                                                                
1          13     Consumer  2014-10-06 00:00:00.000  Standard Class   
2         252     Consumer  2012-10-20 00:00:00.000  Standard Class   
3         193     Consumer  2012-10-20 00:00:00.000  Standard Class   
4          35     Consumer  2012-10-20 00:00:00.000  Standard Class   
5          72     Consumer  2012-10-20 00:00:00.000  Standard 

In [81]:
df

Category           City   Country Customer.ID  \
Row.ID                                                         
1       Office Supplies    Mexico City    Mexico   SC-205753   
2             Furniture  Dos Quebradas  Colombia   KW-165703   
3             Furniture  Dos Quebradas  Colombia   KW-165703   
4       Office Supplies  Dos Quebradas  Colombia   KW-165703   
5       Office Supplies  Dos Quebradas  Colombia   KW-165703   
...                 ...            ...       ...         ...   
51286   Office Supplies       Budapest   Hungary    NF-83853   
51287        Technology       Budapest   Hungary    NF-83853   
51288   Office Supplies       Budapest   Hungary    NF-83853   
51289   Office Supplies       Budapest   Hungary    NF-83853   
51290   Office Supplies        Baghdad      Iraq   SZ-100352   

           Customer.Name  Discount Market               Order.Date  \
Row.ID                                                               
1           Sonia Cooley       0.0  LATAM  2014-10-02 00:00:00.000   
2         Kelly Williams       0.0  LATAM  2012-10-15 00:00:00.000   
3         Kelly Williams       0.0  LATAM  2012-10-15 00:00:00.000   
4         Kelly Williams       0.0  LATAM  2012-10-15 00:00:00.000   
5         Kelly Williams       0.0  LATAM  2012-10-15 00:00:00.000   
...                  ...       ...    ...                      ...   
51286   Natalie Fritzler       0.0   EMEA  2012-09-05 00:00:00.000   
51287   Natalie Fritzler       0.0   EMEA  2012-09-05 00:00:00.000   
51288   Natalie Fritzler       0.0   EMEA  2012-09-05 00:00:00.000   
51289   Natalie Fritzler       0.0   EMEA  2012-09-05 00:00:00.000   
51290         Sam Zeldin       0.0   EMEA  2011-08-25 00:00:00.000   

              Order.ID Order.Priority        Product.ID  \
Row.ID                                                    
1       MX-2014-143658         Medium   OFF-LA-10002782   
2       MX-2012-155047         Medium   FUR-FU-10004015   
3       MX-2012-155047         Medium   FUR-BO-10002352   
4       MX-2012-155047         Medium   OFF-BI-10004428   
5       MX-2012-155047         Medium   OFF-AR-10004594   
...                ...            ...               ...   
51286     HU-2012-7730           High  OFF-AVE-10004570   
51287     HU-2012-7730           High  TEC-LOG-10004419   
51288     HU-2012-7730           High  OFF-BOS-10002705   
51289     HU-2012-7730           High  OFF-ENE-10004132   
51290     IZ-2011-5750         Medium  OFF-ACC-10002343   

                                     Product.Name  Profit  Quantity Region  \
Row.ID                                                                       
1              Hon File Folder Labels, Adjustable    4.56         3  North   
2                            Tenex Clock, Durable   90.72         8  South   
3                    Ikea 3-Shelf Cabinet, Mobile   54.08         2  South   
4                          Cardinal Binder, Clear    4.96         4  South   
5                     Sanford Canvas, Water Color   11.44         2  South   
...                                           ...     ...       ...    ...   
51286                Avery Binder Covers, Economy    3.42         1   EMEA   
51287                   Logitech Flash Drive, USB   18.42         2   EMEA   
51288            Boston Highlighters, Fluorescent   20.88         4   EMEA   
51289   Enermax Computer Printout Paper, 8.5 x 11   33.84         4   EMEA   
51290                  Acco 3-Hole Punch, Durable   12.96         1   EMEA   

        Sales      Segment                Ship.Date       Ship.Mode  \
Row.ID                                                                
1          13     Consumer  2014-10-06 00:00:00.000  Standard Class   
2         252     Consumer  2012-10-20 00:00:00.000  Standard Class   
3         193     Consumer  2012-10-20 00:00:00.000  Standard Class   
4          35     Consumer  2012-10-20 00:00:00.000  Standard Class   
5          72     Consumer  2012-10-20 00:00:00.000  Standard 

Информация во всех остальных столбцах может быть полезной при проведении общей аналитики. Например, столбец «weeknum» может рассказать мне, в какие периоды заказывали чаще, «Discount» покажет мне, как сильно влияет скидка на количество заказов, а «Product.Name» может рассказать мне, какие товары были популярны у покупателей, какие нет.

На данном этапе анализа и предпоготовки таблицы уже были выявлены интересные значения, например: что в таблице есть 53 неделя в связи с интересным календарем на конец 2012 года, данные взяты из выборки 147 стран мира с семи маркетов. Информация была собрана за 4 года продаж, максимальное количество единовременного заказа — 14 штук.
Приступаю к анализу данных на основе измененной мною после первичной проверки таблицы.